sdcm/cluster_gce.py

```diff
@@ -46,27 +46,27 @@ def __init__(self, gce_instance, gce_service, credentials, parent_cluster,  # py
                                       node_prefix=node_prefix,
                                       dc_idx=dc_idx)
         if not cluster.Setup.REUSE_CLUSTER:
-            metadata = []
-            ex_metadata = {'Name': name,
-                           'NodeIndex': node_index,
-                           'NodeType': node_type}
-            ex_metadata = gce_create_metadata(ex_metadata)
+            keep_alive = False
             if cluster.TEST_DURATION >= 24 * 60:
                 self.log.info('Test duration set to %s. '
                               'Tagging node with "keep-alive"',
                               cluster.TEST_DURATION)
-                metadata.append('keep-alive')
+                keep_alive = True
             elif "db" in self.name and cluster.Setup.KEEP_ALIVE_DB_NODES:
                 self.log.info('Keep cluster on failure %s', cluster.Setup.KEEP_ALIVE_DB_NODES)
-                metadata.append('keep-alive')
+                keep_alive = True
             elif "loader" in self.name and cluster.Setup.KEEP_ALIVE_LOADER_NODES:
                 self.log.info('Keep cluster on failure %s', cluster.Setup.KEEP_ALIVE_LOADER_NODES)
-                metadata.append('keep-alive')
+                keep_alive = True
             elif "monitor" in self.name and cluster.Setup.KEEP_ALIVE_MONITOR_NODES:
                 self.log.info('Keep cluster on failure %s', cluster.Setup.KEEP_ALIVE_MONITOR_NODES)
-                metadata.append('keep-alive')
-            self._instance_wait_safe(self._gce_service.ex_set_node_tags,
-                                     self._instance, metadata, ex_metadata)
+                keep_alive = True
+            if keep_alive:
+                self._instance_wait_safe(self._gce_service.ex_set_node_tags, self._instance, ['keep-alive'])
+            self._instance_wait_safe(self._gce_service.ex_set_node_metadata,
```

Comment:
we can't set the metadata here since we need startup script to run and it runs when instance got created.
so the correct place to put this when we run `create_node` in `sdcm.cluster_gce.GCECluster#_create_instance`